vault/seal_autoseal.go

```diff
@@ -499,3 +506,52 @@ func (d *autoSeal) migrateRecoveryConfig(ctx context.Context) error {
 
 	return nil
 }
+
+func (d *autoSeal) StartHealthCheck() {
+	d.healthCheck = time.NewTicker(sealHeathTestInterval)
+	d.healthCheckStop = make(chan struct{})
+	go func() {
+		lastTestOk := true
+		lastSeenOk := time.Now()
+		for {
+			select {
+			case <-d.healthCheckStop:
+				d.healthCheck.Stop()
```

Comment:
I'm probably being paranoid here but if we get multiple healthCheckStop we will get a nil deference error. Is it worth adding a quick test and return if d.healthCheck is nil?

Reply:
nvm the case statement will never fire.

Reply:
yes it is, thanks.